enumerable.rb

```diff
@@ -0,0 +1,126 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    for i in self
+      yield(i)
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    for i in 0...size
+      yield(to_a[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    if is_a? Hash
+      helper = {}
+      my_each { |k, v| helper[k] = v if yield(k, v) }
+    else
+      helper = []
+      my_each { |i| helper << i if yield(i) }
+    end
+    helper
+  end
+
+  def my_all?(args = nil)
+    query_result = true
+    if args.nil? && !block_given?
+      my_each { |i| query_result = false if !i || i.nil? }
+    elsif args.nil?
+      my_each { |i| query_result = false unless yield(i) }
+    elsif args.is_a? Regexp
+      my_each { |i| query_result = false unless i.match(args) }
+    elsif args.is_a? Module
+      my_each { |i| query_result = false unless i.is_a?(args) }
+    else
+      my_each { |i| query_result = false unless i == args }
+    end
+    query_result
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    count = size if !block_given? && arg.nil?
+    my_each { |i| count += 1 unless i != arg } if arg
+    my_each { |i| count += 1 if yield(i) } if block_given?
+    puts count
+  end
```

Comment:
- [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't work when no argument or block is given. It should return the count of all elements inside the innumerable. Right now it's returning nil.

- Your method doesn't work when a number is given as an argument `(some_enumerable.my_count(5)).` It should return the count of all elements that match the argument. Right now it's returning nil.

- Your method doesn't work when a block is given. It should return the count of all elements that match the block. Right now it's returning nil.

Reply:
Thanks for the suggestions. I checked it and I disagree with this review, as it is working as it should be. I only removed 'puts' from the last line.